### PR TITLE
Feature/ap 13

### PR DESCRIPTION
**Pull request by rdhawladar**

### CODE REVIEW

File: internal/app/carrier/repository.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	"parcel-service/internal/app/model"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq"
+)
+
+// sql query and error
+const (
+	errUniqueViolation = pq.ErrorCode("23505")
+	insertCarrierQuery = `INSERT INTO carrier_request (carrier_id, parcel_id) VALUES ($1, $2)`
```
Comment (shuza) [inline internal/app/carrier/repository.go:15]: stats is required. how status will inserted at there?

File: internal/app/carrier/repository_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		}
+
+		sqlxDB := sqlx.NewDb(db, "sqlmock")
+		m.ExpectExec("INSERT INTO carrier_request (.+) VALUES (.+)").
+			WithArgs(carrierRequest.CarrierID, carrierRequest.ParcelID).
+			WillReturnResult(sqlmock.NewResult(1, 1))
+
+		repo := NewRepository(sqlxDB)
+		err := repo.InsertCarrierRequest(context.Background(), carrierRequest)
+		assert.True(t, errors.Is(err, nil))
```
Comment (shuza) [inline internal/app/carrier/repository_test.go]: better to use `nil` assert

File: internal/app/carrier/repository_test.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		}
+
+		sqlxDB := sqlx.NewDb(db, "sqlmock")
+		m.ExpectExec("INSERT INTO carrier_requests (.+) VALUES (.+)").
+			WithArgs().
+			WillReturnError(errors.New("sql-error"))
+
+		repo := NewRepository(sqlxDB)
+		err := repo.InsertCarrierRequest(context.Background(), carrierRequest)
+		assert.NotNil(t, err)
```
Comment (shuza) [inline internal/app/carrier/repository_test.go]: should check error as you know the error

**Review by shuza — CHANGES_REQUESTED** [review]
it should not contain parcel create api code as we have different task for that

File: internal/app/carrier/repository.go (hunk 1 above)
Comment (rdhawladar) [reply internal/app/carrier/repository.go:15]: @shuza  status value is default 1 in DB. No need to insert default value from code,  `status INT NOT NULL DEFAULT 1`

File: internal/app/server/handler.go [hunk 4]
```diff
@@ -33,3 +35,36 @@ func (s *server) createParcel(w http.ResponseWriter, r *http.Request) {
...
+		ErrInvalidEntityResponse(w, "Invalid Input", err)
+		return
+	}
+
+	if err := s.carrierService.NewCarrierRequest(r.Context(), data); err != nil {
+		if errors.Is(err, model.ErrInvalid) {
+			ErrInvalidEntityResponse(w, "invalid Request", err)
+			return
+		}
+		log.Error().Err(err).Msgf("[parcel/{id}/request] failed to add new carrier request: %v", err)
```
Comment (shuza) [inline internal/app/server/handler.go]: better to use [methodName] in log as other methods will not have routes

**Review by shuza — APPROVED** [review]
Better to use [methodNmae] in log otherwise lgtm

**Pull request merged**